### PR TITLE
[VLM] Support Piecewise CUDA Graph for Qwen3-Omni-MOE

### PR DESCRIPTION
**Pull request by yuan-luo**
## Motivation

Follow up https://github.com/sgl-project/sglang/pull/14155, supports Qwen3-Omni-MOE model piecewise CUDA graph.

The PCG framework doesn't recognize "thinker" layer. This PR fixes this issue and supports Qwen3 VLM family PCG.
```
Qwen3OmniMoeForConditionalGeneration(
  (thinker): Qwen3OmniMoeThinkerForConditionalGeneration(
    (visual): Qwen3OmniMoeVisionEncoder(
......
    )
    (model): Qwen3MoeLLMModel(
      (embed_tokens): VocabParallelEmbedding(num_embeddings=38016, embedding_dim=2048, org_vocab_size=152064, num_embeddings_padded=152064, tp_size=4)
.....
    )
  )
)
```

```
➜  sglang_dev2 git:(main) ✗ python -m pytest ./test/srt/test_piecewise_cuda_graph.py::TestPiecewiseCudaGraphQwen3OmniMOE
================================================================================================================ test session starts ================================================================================================================
platform linux -- Python 3.12.12, pytest-8.4.2, pluggy-1.6.0
rootdir: /sgl-workspace/workspace/sglang_dev2/test
configfile: pytest.ini
plugins: anyio-4.11.0, typeguard-4.4.4
collected 1 item

test/srt/test_piecewise_cuda_graph.py .                                                                                                                                                                                                       [100%]

================================================================================================================= warnings summary ==================================================================================================================
../../../usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1474
  /usr/local/lib/python3.12/dist-packages/_pytest/config/__init__.py:1474: PytestConfigWarning: Unknown config option: asyncio_mode

    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

../../../usr/local/lib/python3.12/dist-packages/sglang/srt/sampling/sampling_params.py:17
  /usr/local/lib/python3.12/dist-packages/sglang/srt/sampling/sampling_params.py:17: DeprecationWarning: module 'sre_parse' is deprecated
    import sre_parse

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================================================================== 1 passed, 4 warnings in 195.51s (0:03:15) =====================================================================================================

```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/model_executor/model_runner.py [hunk 1]
```diff
@@ -247,6 +248,12 @@ def add_chunked_prefix_cache_attention_backend(backend_name):
     torch_npu.npu.set_compile_mode(jit_compile=False)
 
 
+def resolve_language_model(model: nn.Module) -> nn.Module:
+    if hasattr(model, "thinker") and hasattr(model.thinker, "model"):
```
Comment (ByronHsu) [inline python/sglang/srt/model_executor/model_runner.py]: can we do different resolve based on **model name**? checking `thinker` attribute on model seems a bit hacky

Comment (yuan-luo) [reply python/sglang/srt/model_executor/model_runner.py]: Revised to model name.

**Review by ByronHsu — APPROVED** [review]

**Review by BBuf — APPROVED** [review]

File: test/srt/run_suite.py [hunk 2]
```diff
@@ -158,6 +157,7 @@
         TestFile("test_local_attn.py", 411),
         TestFile("test_multi_instance_release_memory_occupation.py", 64),
         TestFile("test_pp_single_node.py", 481),
+        TestFile("test_piecewise_cuda_graph.py", 1200),
```
Comment (merrymercy) [inline test/srt/run_suite.py:160]: This is a very bad change. It breaks multiple principles and leads to CI timeout. Here are the principles:
- For tests that do not need many GPUs, do not run on multi-GPU runners. Split the test files and only run large tests with mulit-gpu runners.
- A single test file should only run < 500s. Otherwise, please split the big file into smaller ones.

The mulit-gpu runners are very precious resources. You should not waste it.
In your case, you should create a new file instead of moving all 1-gpu tests to 4-gpu runners.
Also, can we run `Qwen3OmniMOE` with TP=2 or TP=1?

Comment (yuan-luo) [reply test/srt/run_suite.py:160]: Sorry for my fault. I changed to tp=1 and the case passed. It cost 219.13s. Given the original timeout is 850s, I set 1200s as leaving some buffer. Creating a new PR and move it into GPU=1 test suite.
```
TestFile("test_piecewise_cuda_graph.py", 850),
```

```
$python -m pytest ./test/srt/test_piecewise_cuda_graph.py::TestPiecewiseCudaGraphQwen3OmniMOE -vvvv
============================================================================================= test session starts =============================================================================================
platform linux -- Python 3.10.13, pytest-8.3.5, pluggy-1.5.0 -- /opt/conda/bin/python
cachedir: .pytest_cache
rootdir: /root/luoyuan.luo/workspace/sglang_dev/test
configfile: pytest.ini
plugins: anyio-4.12.0
collected 1 item                                                                                                                                                                                              

test/srt/test_piecewise_cuda_graph.py::TestPiecewiseCudaGraphQwen3OmniMOE::test_gsm8k_accuracy PASSED                                                                                                   [100%]

============================================================================================== warnings summary ===============================================================================================
../../../../opt/conda/lib/python3.10/site-packages/_pytest/config/__init__.py:1441
  /opt/conda/lib/python3.10/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: asyncio_mode
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../../../opt/conda/lib/python3.10/site-packages/torch/cuda/__init__.py:63
  /opt/conda/lib/python3.10/site-packages/torch/cuda/__init__.py:63: FutureWarning: The pynvml package is deprecated. Please install nvidia-ml-py instead. If you did not install pynvml directly, please report this to the maintainers of the package that installed pynvml for you.
    import pynvml  # type: ignore[import]

<frozen importlib._bootstrap>:241
  <frozen importlib._bootstrap>:241: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:241
  <frozen importlib._bootstrap>:241: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== 1 passed, 4 warnings in 219.13s (0:03:39) ==================================================================================

```

Comment (alisonshao) [reply test/srt/run_suite.py:160]: hi, i created a fix at: #15290

Comment (yuan-luo) [reply test/srt/run_suite.py:160]: @alisonshao Thanks a lot.
